dmutils/content_loader.py

```diff
@@ -111,15 +75,58 @@ def _question_should_be_shown(self, dependencies, service_data):
         return True
 
 
-class YAMLLoader(object):
-
-    def __init__(self):
+class ContentLoader(object):
+    def __init__(self, manifest, content_directory):
         self._cache = {}
+        self._directory = content_directory
+
+        manifest_sections = read_yaml(manifest)
+
+        self._questions = {
+            q: self._load_question(q)
+            for section in manifest_sections
+            for q in section["questions"]
```

Comment:
Nested comprehensions make my brain hurt. This one isn't so bad because the variable names make it fairly obvious but I did have to stop and have a think about it.